### ISSUE TITLE
Support for version 2.021

### ISSUE THREAD

**Issue by binaryguru** [Aktanusa/CookieMonster#257]
Loading the script locks the game up.

**Comment by darkenvy** [comment]
I have a merge request in place that fixes the crashes. Does not support new features, but makes it so it works with the game again.

Until the merge request is fixed, you can load it from here: https://raw.githubusercontent.com/darkenvy/CookieMonster/2.021-hotfix/CookieMonster.js

You wont be able to use `Game.load()` however as it throws a CORS issue. This is why Aktanusa hosts the script on their own website. So you will have to copy and paste the entire script into your console window bleh!

Or just make a tampermonkey script. The below does work. I tested it.

```
// ==UserScript==
// @name Cookie Monster
// @namespace Cookie
// @include http://orteil.dashnet.org/cookieclicker/
// @include https://orteil.dashnet.org/cookieclicker/
// @version 1
// @grant none
// @description Loads the CookieMonster helper tool for CookieClicker Idle game
// ==/UserScript==

(function() {
    var checkReady = setInterval(function() {
        if (typeof Game.ready !== 'undefined' && Game.ready) {
            loadCMManually();
            clearInterval(checkReady);
        }
    }, 1000);
}());

function loadCMManually() {
 // paste the contents of https://raw.githubusercontent.com/darkenvy/CookieMonster/2.021-hotfix/CookieMonster.js here
}
```

**Comment by Aktanusa** [comment]
Yep, working on it...  Thanks @darkenvy, as I said in your pull request, I'll take a look at your changes and will give you credit.

**Comment by normann1974** [comment]
What's the plan for fortune cookies? It would be nice with a sound being played.

**Comment by Piruparka** [comment]
> I have a merge request in place that fixes the crashes. Does not support new features, but makes it so it works with the game again.
> 
> Until the merge request is fixed, you can load it from here: https://raw.githubusercontent.com/darkenvy/CookieMonster/2.021-hotfix/CookieMonster.js
> 
> You wont be able to use `Game.load()` however as it throws a CORS issue. This is why Aktanusa hosts the script on their own website. So you will have to copy and paste the entire script into your console window bleh!
> 
> Or just make a tampermonkey script. The below does work. I tested it. Just be sure to paste the script where I left the note.
> 
> ```
> // ==UserScript==
> // @name Cookie Monster
> // @namespace Cookie
> // @include http://orteil.dashnet.org/cookieclicker/
> // @include https://orteil.dashnet.org/cookieclicker/
> // @version 1
> // @grant none
> // @description Loads the CookieMonster helper tool for CookieClicker Idle game
> // ==/UserScript==
> 
> (function() {
>     var checkReady = setInterval(function() {
>         if (typeof Game.ready !== 'undefined' && Game.ready) {
>             loadCMManually();
>             clearInterval(checkReady);
>         }
>     }, 1000);
> }());
> 
> function loadCMManually() {
>  // ###### paste the contents of https://raw.githubusercontent.com/darkenvy/CookieMonster/2.021-hotfix/CookieMonster.js here #######
> }
> ```
> 
> **UPDATE:** You can use `Game.LoadMod('https://cdn.statically.io/gh/darkenvy/CookieMonster/2.021-hotfix/CookieMonster.js');` as @storgall pointed out :). Thanks Storgall!

Weird question but how do I use this?

**Comment by normann1974** [comment]
It doesn't seem to work in any useful way. All bonus incomes and payback periods are negative.

**Comment by Piruparka** [comment]
I made it work now and it's ok.

I created a bookmark aiming at

`javascript:(function() { Game.LoadMod('https://cdn.statically.io/gh/darkenvy/CookieMonster/2.021-hotfix/CookieMonster.js'); }());`

and clicked then while the game is running

**Comment by Batman42** [comment]
I've tried all the bookmarks above. None seem to work in the way they are supposed to. Payback periods are all negative and aren't calculating properly.

**Comment by Piruparka** [comment]
Try the one I pasted above, I'm running it rn and it's running ok

**Comment by Fataky** [comment]
> Try the one I pasted above, I'm running it rn and it's running ok

It works perfectly for me. Thanks!

**Comment by Aktanusa** [comment]
Thanks @Piruparka for helping out while waiting for me.  I understand a lot of people are waiting on me, and I'm trying to get on it ASAP.

@normann1974 Yes, it has been requested already.  I'll try to add it also if it's not too complicated, or else it will be added later.

**Comment by Batman42** [comment]
> Try the one I pasted above, I'm running it rn and it's running ok

I've tried that one, I think it was working until I bought the new executive kitten upgrade then it went back to not working

**Comment by Azmaeus** [comment]
The negative numbers are apparently a result of the new Dragon Aura, per #256 - however I switched to a different aura and am still experiencing this...?

**Comment by Azmaeus** [comment]
> @normann1974 Yes, it has been requested already. I'll try to add it also if it's not too complicated, or else it will be added later.

@Aktanusa I can't find the request, was hoping to make comments. If it's public I'd appreciate a link, otherwise I'm working on a PR, hopefully I'll have something soon.

**Comment by Aktanusa** [comment]
@Azmaeus It was an informal request on reddit.  I guess I should make an issue, or get someone to do it, lol.  Tomorrow, I'll make an issue if no one does.

**Comment by mocnik** [comment]
I have fixed some issues with payback period calculation (add new kitten and fortune upgrades, fix building fortune bonuses) in https://github.com/mocnik/CookieMonster/tree/2.021-hotfix, feel free to use it.

**Comment by Aktanusa** [comment]
@mocnik Thanks, it will help me.  I have started to look at the changes last night, and so far, it doesn't look to bad.  I will have more progress tonight.

**Comment by moguling** [comment]
@mocnik Is there a way to use this version in a bookmarklet?

**Comment by Doogie2K** [comment]
Thanks for doing this @Aktanusa, and thanks for the hotfix, @darkenvy. I appreciate the work that goes into this mod, and that y'all have lives that don't necessarily revolve around a silly browser game. ;)

**Comment by Zatch97** [comment]
Hi guys, thanks for your great work, just wondering if it's normal with the hotfixes that I don't have access to the Cookie Monster panel that was replacing the Info panel?

**Comment by Antwelm** [comment]
> Hi guys, thanks for your great work, just wondering if it's normal with the hotfixes that I don't have access to the Cookie Monster panel that was replacing the Info panel?

Stats & Options show Monster tweaks here, info-tab shows normal CC info.

**Comment by shiioo** [comment]
@moguling yes, just follow previous recommendations on this same thread:

`javascript: (function () { Game.LoadMod('https://cdn.statically.io/gh/mocnik/CookieMonster/2.021-hotfix/CookieMonster.js'); }());`

**Comment by Aktanusa** [comment]
Fixed in Version 2.021.1
